lib/msal-common/src/response/ResponseHandler.ts

```diff
@@ -51,6 +54,50 @@ export class ResponseHandler {
             tenantId: idTokenObj.claims.tid,
         };
     }
+
+    private saveToken(originalTokenResponse: TokenResponse, authority: string, serverTokenResponse: ServerAuthorizationTokenResponse, clientInfo: ClientInfo): TokenResponse {
+        const tokenResponse = { ...originalTokenResponse };
+        // Set consented scopes in response
+        const requestScopes = ScopeSet.fromString(serverTokenResponse.scope, this.clientId, false);
+        tokenResponse.scopes = requestScopes.asArray();
+
+        const accessTokenCacheItems = this.cacheManager.getAllAccessTokens(this.clientId, authority);
+
+        for (let i = 0; i < accessTokenCacheItems.length; i++) {
+            const accessTokenCacheItem = accessTokenCacheItems[i];
+            if (accessTokenCacheItem.key.homeAccountIdentifier === tokenResponse.account.homeAccountIdentifier) {
```

Comment:
Instead of using `tokenResponse`, you can use `originalTokenResponse`.

Reply:
done.